adr/0002-quarkus-rest.adoc

```diff
@@ -0,0 +1,37 @@
+= Rename RESTEasy Reactive extensions to Quarkus REST
+
+* Status: proposed
+* Date: 2023-10-17
+
+== Context and Problem Statement
+
+RESTEasy Reactive is the preferred REST stack in Quarkus, however in the almost 3 years since its release, we have seen
+folks of all experience misunderstand its capabilities and being led to believe that it forces reactive programming on them.
+
+We have tried to clarify the situation with blog posts, talks and on the various communication channels, but there seems to be
+no escaping the fact that "reactive" in the name primes people in a non-desirable way.
+
+The proposal is to rename the Quarkus extensions only (the independent project itself would continue to be called RESTEasy Reactive)
+to `Quarkus REST`.
+
+== Considered options
+
+We can continue with the current situation, but that feels less than ideal given the recurrence of the problem.
+
+== Decision
+
+* Rename all `quarkus-resteasy-reactive*` maven modules to `quarkus-rest*`
+* No Java code changes should be made - all RESTEasy Reactive code should continue to reside in the `org.jboss.resteasy.reactive.*` packages
+* Introduce Maven relocations to allow for users not updating their poms
+* Introduce a recipe that the Quarkus CLI can use during update
+* Update quickstarts
+* Update codestarts
+* Write a new blog post explaining why we made the change
+* Update old blog posts to mention the new name
+* Update prominent Quarkus demo application and workshops
+
+== Consequences
+
+* On the plus side, conveying the actual capabilities (blocking, non-blocking, Virtual Threads) of the REST stack should become easier.
+** Furthermore, the new name aligns quite nicely with Java EE's Jakarta REST
+* On the minus side, renaming is always confusing for users.
```

Comment:
I would add it paves the road to other renamings:

- reactive messaging -> quarkus messaging
- reactive routes -> quarkus HTTP routes (I think adding HTTP would avoid confusion)

Reply:
I made the proposal more geneal